### ISSUE TITLE
It's hard to print an Uri

### ISSUE THREAD

**Issue by martijnhoekstra** [Spinoco/protocol#12]
Uri has no facilities to render to string. That's unfortunate both for debugging and serialization

**Comment by fommil** [comment]
https://github.com/Spinoco/fs2-http/issues/15 moved here

```
Uri.codec.encode(uri).require
```

gets us a `ByteVector` but I don't know how to get the string out from that.

I also do this https://gitlab.com/fommil/drone-dynamic-agents/blob/master/src/main/scala/http/encoding/urlencoded.scala which is *really* heavyweight.

**Comment by fommil** [comment]
I asked here how to do that :point_up: [November 13, 2017 7:00 PM](https://gitter.im/scodec/scodec?at=5a09ebcd982ea2653f7495a2) lets see if anybody knows.

**Comment by fommil** [comment]
```scala
  @SuppressWarnings(Array("org.wartremover.warts.EitherProjectionPartial"))
  def stringify(uri: Uri): String =
    Uri.codec
      .encode(uri)
      .require
      .decodeUtf8
      .right
      .get
```

which seems to assume that it encodes and decodes.

**Comment by pjrt** [comment]
Very odd hole since Path has a `stringify`.

**Comment by pchlupacek** [comment]
Fixed in 0.3.0 -> Uri.stringify
